core/blockchain.go

```diff
@@ -499,11 +510,31 @@ func (bc *BlockChain) cacheReceipts(hash common.Hash, receipts types.Receipts) {
 	bc.receiptsCache.Add(hash, receipts)
 }
 
-func (bc *BlockChain) cacheDiffLayer(diffLayer *types.DiffLayer) {
+func (bc *BlockChain) cacheDiffLayer(diffLayer *types.DiffLayer, diffLayerCh chan struct{}) {
+	sort.SliceStable(diffLayer.Codes, func(i, j int) bool {
```

Comment:
dd some comments to explain why do sorting here?

Reply:
added